Tasks/JenkinsQueueJobV2/jobsearch.ts

```diff
@@ -43,23 +43,23 @@ export class JobSearch {
         if (!thisSearch.Initialized) { //only initialize once
             const apiTaskUrl: string = util.addUrlSegment(thisSearch.taskUrl, '/api/json?tree=downstreamProjects[name,url,color],lastBuild[number]');
             tl.debug('getting job task URL:' + apiTaskUrl);
-            request.get({ url: apiTaskUrl, strictSSL: thisSearch.queue.TaskOptions.strictSSL }, function requestCallBack(err, httpResponse, body) {
+
+            var request = new WebRequest();
+            request.uri = apiTaskUrl;
+            request.method = "GET";
+            request.headers = {
+                "Content-Type": "application/json; charset=utf-8",
+                "Authorization": 'Basic ' + new Buffer(thisSearch.queue.TaskOptions.username + ':' + thisSearch.queue.TaskOptions.password).toString('base64')
+            };
+        
+            var response = sendRequest(request).then(async (response: WebResponse) => {
                 if (!thisSearch.Initialized) { // only initialize once
-                    if (err) {
-                        if (err.code == 'ECONNRESET') {
-                            tl.debug(err);
-                            // resolve but do not initialize -- a job will trigger this again
-                            defer.resolve(null);
-                        } else {
-                            defer.reject(err);
-                        }
-                    } else if (httpResponse.statusCode !== 200) {
-                        defer.reject(util.getFullErrorMessage(httpResponse, 'Unable to retrieve job: ' + thisSearch.identifier));
+                    if (response.statusCode !== 200) {
+                        defer.reject(util.getFullErrorMessage(response, 'Unable to retrieve job: ' + thisSearch.identifier));
                     } else {
-                        const parsedBody: any = JSON.parse(body);
```

Comment:
Parsing is now done in the webclient with the exception of when using the sendRequestNoParsing method.